### PR TITLE
Fix #76 :Responsiveness of BusDetails Card  component

### LINKED ISSUE #76: BusDetailsCard component is not responsive for tablets & small laptops screen.
### Describe the bug

The `BusDetailCard` component is overflowing out of its designated area. Also, the Divider becomes invisible in `768px-1024px`, which is unexpected behavior according to the written code logic.

Screenshot from 2024-06-01 12-58-22

### Steps To Reproduce

1. Click the see schedule button
2. Click on Book ticket

### Expected Behavior

- The `divider` should be visible for small laptop screen sizes.
- The `BusDetailsCard` component should be responsive.

### Screenshots/Videos

_No response_

### What device are you using?

Desktop

### Operating System

Linux

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Zapper9982**
## Overview

1. This PR fixes or fixes part of #76 .
2. This PR does the following: The PR fixes the responsiveness of the BusDetailsCard Component for small laptop screens and Tablets .

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
- [x] The PR is made from a branch that's **not** called "main/master".

## Proof that changes are correct

Screenshot 2024-06-02 at 3 08 38 PM
Screenshot 2024-06-02 at 3 08 53 PM
Screenshot 2024-06-02 at 3 09 12 PM

## PR Pointers

- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL".
- Never force push. If you do, your PR will be closed.

### CODE REVIEW

File: frontend/src/components/BusDetailsCard/index.tsx [hunk 1]
```diff
@@ -19,34 +19,37 @@ function BusDetailsCard() {
...
   const destination = useOrderStore(state => state.destination);
+
   return (
     <>
       <Box
         sx={{
           display: 'flex',
-          alignItems: 'center',
-          gap: {xs: '10px', md: '2rem'},
+          flexDirection: {xs: 'row', sm: 'row'},
```
Comment (Pratik2026) [inline frontend/src/components/BusDetailsCard/index.tsx:28]: we don't need this as flex-direction is row by default.

File: frontend/src/components/BusDetailsCard/index.tsx [hunk 2]
```diff
@@ -19,34 +19,37 @@ function BusDetailsCard() {
...
+          gap: {xs: '10px', sm: '2px', md: '10px'},
+          margin: {xs: '0', md: '0 0.5rem'},
         }}
       >
         <Box
           sx={{
             display: 'flex',
             alignItems: 'center',
-            gap: {xs: '5px', md: '10px'},
+            gap: {xs: '5px', md: '5px'},
```
Comment (Pratik2026) [inline frontend/src/components/BusDetailsCard/index.tsx:38]: let's write it as `gap: '5px'` as both produce the same result, so we should opt the one having better readability.

File: frontend/src/components/BusDetailsCard/index.tsx [hunk 3]
```diff
@@ -19,34 +19,37 @@ function BusDetailsCard() {
...
+
   return (
     <>
       <Box
         sx={{
           display: 'flex',
-          alignItems: 'center',
-          gap: {xs: '10px', md: '2rem'},
+          flexDirection: {xs: 'row', sm: 'row'},
+          alignItems: {xs: 'center', sm: 'center'},
```
Comment (Pratik2026) [inline frontend/src/components/BusDetailsCard/index.tsx:29]: Let's write it as `alignItems: 'center'`

File: frontend/src/components/BusDetailsCard/index.tsx [hunk 4]
```diff
@@ -58,21 +61,21 @@ function BusDetailsCard() {
           sx={{
             display: 'flex',
             alignItems: 'center',
-            gap: {xs: '5px', md: '10px'},
+            gap: {xs: '5px', md: '5px'},
```
Comment (Pratik2026) [inline frontend/src/components/BusDetailsCard/index.tsx:64]: do the same as mentioned previously.

**Comment by Pratik2026** [comment]
@Zapper9982, I've noticed that some unnecessary breakpoints have been applied. I've pointed out these instances in the review comments. 

Always apply breakpoints only when they are required. This helps keep the code clean and easier to maintain. Other than that the code is working as expected. Just make the suggested changes then we'll be all set to proceed.

**Comment by Zapper9982** [comment]
I have made the required changes to the code , as mentioned  .

**Comment by Pratik2026** [comment]
@Zapper9982 LGTM.

**Comment by Pratik2026** [comment]
merged, thanks for contributing @Zapper9982

**Pull request merged**